channeldb/kvdb/interface.go

```diff
@@ -0,0 +1,90 @@
+package kvdb
+
+import (
+	"github.com/btcsuite/btcwallet/walletdb"
+	_ "github.com/btcsuite/btcwallet/walletdb/bdb" // Import to register backend.
+)
+
+// Update opens a database read/write transaction and executes the function f
+// with the transaction passed as a parameter. After f exits, if f did not
+// error, the transaction is committed. Otherwise, if f did error, the
+// transaction is rolled back. If the rollback fails, the original error
+// returned by f is still returned. If the commit fails, the commit error is
+// returned.
+var Update = walletdb.Update
+
+// View opens a database read transaction and executes the function f with the
+// transaction passed as a parameter. After f exits, the transaction is rolled
+// back. If f errors, its error is returned, not a rollback error (if any
+// occur).
+var View = walletdb.View
+
+// Batch is identical to the Update call, but it attempts to combine several
+// individual Update transactions into a single write database transaction on
+// an optimistic basis. This only has benefits if multiple goroutines call
+// Batch.
+var Batch = walletdb.Batch
```

Comment:
This may be problematic as `etcd` does not support chaning the same key in the same transaction while `bbolt` is ok with that.

Reply:
Chaining the same key? This is more about combining several write transactions into one. On the implementation side, if there isn't a clear mapping, then we can simply have it just route through to `Update`.